### DISCUSSION TITLE
Basic Question on Non-Flake Install

### DISCUSSION THREAD

**Discussion by gigibytes** [nix-community/nixvim#783] [Q&A] [answered]
Hi. Sorry for the basic question, but I am not sure exactly how the given code for the 'Without flakes' install option should be added to a configuration.nix file. (referencing this: https://github.com/nix-community/nixvim#without-flakes)

I threw it in my configuration.nix and got errors related to unexpected brackets, it seems like I shouldn't have just dropped it in there the way I did (right after my user profile config).

```nix
  # Define a user account. Don't forget to set a password with ‘passwd’.
  users.users.gigi = {
 ...
  };

  # Nixvim Configuration
  { pkgs, lib, ... }:
  let
  nixvim = import (builtins.fetchGit {
    url = "https://github.com/nix-community/nixvim";
  });
  in
  {
  imports = [
    # For NixOS
    nixvim.nixosModules.nixvim
  ];

  programs.nixvim.enable = true;
  plugins.which-key.enable = true;
  colorschemes.tokyonight.enable = true;
  }
```

Does anyone have an example of a 'real' non-flake install or have pointers on how I should be adding it to my configuration.nix? I have only seen flake-based configurations.

Thanks for any help.

**Comment by GaetanLepage** [comment, marked as answer]
Hello !
Indeed, you should not just put it like this.
The `{ pkgs, lib, ... }:` should be at the beginning of the file.
Then you can just add the rest within your `configuration.nix` file.

**Reply by GaetanLepage** [reply]
Here is what it should more or less look like.

`configuration.nix`
```nix
{ config, lib, pkgs, ... }:
let
  nixvim = import (builtins.fetchGit {
    url = "https://github.com/nix-community/nixvim";
  });
in

{
  imports =
    [ # Include the results of the hardware scan.
      ./hardware-configuration.nix
      nixvim.nixosModules.nixvim
    ];

  users.users.gigi = {
    ...
  };

  # Nixvim configuration
  programs.nixvim = {
    enable = true;

    plugins.which-key.enable = true;
    colorschemes.tokyonight.enable = true;
  };


  system.stateVersion = "24.05"; # Did you read the comment?

}
```

**Reply by gigibytes** [reply]
thanks, really appreciate it.

**Reply by GaetanLepage** [reply]
Glad that it works !
In case you have other questions, feel free to join the [matrix chat](https://matrix.to/#/#nixvim:matrix.org) :)
